app/services/puppet_fact_parser.rb

```diff
@@ -166,6 +166,69 @@ def bios
     {:vendor => facts.dig('dmi', 'bios', 'vendor') || facts['bios_vendor'], :version => facts.dig('dmi', 'bios', 'version') || facts['bios_version'], :release_date => facts.dig('dmi', 'bios', 'release_date') || facts['bios_release_date']}
   end
 
+  # AWS cloud billing fields
+  def aws_account_id
+    facts['aws_account_id']
+  end
+
+  def aws_billing_products
+    facts['aws_billing_products']
+  end
+
+  def aws_instance_id
+    facts['aws_instance_id']
+  end
+
+  def aws_instance_type
+    facts['aws_instance_type']
+  end
+
+  def aws_marketplace_product_codes
+    facts['aws_marketplace_product_codes']
+  end
+
+  def aws_region
+    facts['aws_region']
+  end
+
+  # Azure cloud billing fields
+  def azure_instance_id
+    facts['azure_instance_id']
+  end
+
+  def azure_offer
+    facts['azure_offer']
+  end
+
+  def azure_sku
+    facts['azure_sku']
+  end
+
+  def azure_subscription_id
+    facts['azure_subscription_id']
+  end
+
+  # GCP cloud billing fields
```

Comment:
And for GCP there's the `gce` fact that includes the metadata: https://github.com/puppetlabs/facter/blob/c22260225d420e72240c2b77725794041dcade6a/lib/schema/facter.yaml#L416-L425